app/audit/crypto.ts

```diff
@@ -93,41 +104,44 @@ export const decryptWithPrivateKey = (
  * with the given RSA public key, and returns both values in hex encoding separated by ASYM_SEPARATOR.
  *
  * @param data the plaintext data to encrypt
- * @param publicKey an RSA public key in PEM format
+ * @param scheme object containing the RSA public key in PEM format and a version prefix
  * @return {string} hex-encoded encrypted symmetric key + ASYM_SEPARATOR + hex-encoded encrypted data
  */
-export const encryptData = (data: string, publicKey: Buffer) => {
+export const encryptData = (data: string, scheme: AuditLogScheme) => {
+  const publicKey = Buffer.from(scheme.key);
```

Comment:
Shouldn't we add an invariant here? Otherwise, the key might not be set in the env and we would not get an error (unless there is one in the crypto library?

Reply:
Now I am thinking that whether we could already add a function for re-encryption?

Reply:
> Shouldn't we add an invariant here? Otherwise, the key might not be set in the env and we would not get an error (unless there is one in the crypto library?

The env var is mandatory and there's already a fail-fast mechanism in place (through dotenv-safe) in case it's missing. The crypto library would also throw an error (however, by that point it would be too late since it wouldn't abort whatever operation that triggered the audit log storage).

> Now I am thinking that whether we could already add a function for re-encryption?

The purpose of this PR was to switch to the new table ASAP so that there'd be less data to migrate. Moreover, I wanted to keep the scope tight so that it'd be easier to review and to spot issues. All migration logic is therefore left out on purpose (see PR description).

Reply:
> The purpose of this PR was to switch to the new table ASAP so that there'd be less data to migrate. Moreover, I wanted to keep the scope tight so that it'd be easier to review and to spot issues. All migration logic is therefore left out on purpose (see PR description).

Ok, I missed that during the review.

> The env var is mandatory and there's already a fail-fast mechanism in place (through dotenv-safe) in case it's missing. The crypto library would also throw an error (however, by that point it would be too late since it wouldn't abort whatever operation that triggered the audit log storage).

We had the invariant in-place in the old version (at another place). That's why I'm wondering whether it would now make sense to have this here instead of deleting it

Reply:
Are you referring to the one in `encryptAuditLogData`? The only purpose of that one was to satisfy typescript since otherwise the compiler would complain about the `Buffer.from()` call in the next line. This was semantically redundant though since dotenv-safe already guarantees that the variable is set during app startup. In the rewrite this need was eliminated by the type assertion in `AuditLogScheme`.

Reply:
Yes, that was where I was coming from. Thanks for the explanation 👍